src/charm.py

```diff
@@ -47,38 +54,119 @@ def __init__(self, *args):
         self.mongodb = DatabaseRequires(self, relation_name="mongodb", database_name=self.database)
         self.framework.observe(self.mongodb.on.database_created, self._on_database_created)
 
-        self._on_config_changed(None)
+        self.kafka = KafkaRequires(
+            self, relation_name="kafka", topic=self.topic, extra_user_roles=self.extra_user_roles
+        )
+        self.framework.observe(self.kafka.on.topic_created, self._on_topic_created)
 
     def _on_config_changed(self, event):
-        self.database = self.model.config.get("database", "")
-        if self.database == "":
-            self.unit.status = BlockedStatus("The database name is not specified.")
-        elif event:
-            self.unit.status = ActiveStatus(f"database: {self.database}")
+        """Handle on config changed event."""
+        # Read new parameters
+        new_database_name = self.model.config.get("database-name", "")
+        new_topic = self.model.config.get("topic-name", "")
+        new_extra_user_roles = self.model.config.get("extra-user-roles", "")
+
+        # check the
+        if self.database == "" or not self.is_database_related:
+            self.set_secret("app", "database", new_database_name)
+        elif self.database != new_database_name:
+            self.unit.status = WaitingStatus(
```

Comment:
Sorry for being pedantic, but I really don't think it is a WaitingStatus. According to docs:

waiting =>  The charm is alive and well, and is waiting for some event to occur. No human intervention required.

I believe it makes more sense the blocked here:



blocked  =>  The charm is stuck. Human attention/intervention is required.